### ISSUE TITLE
Blitzortung. Detected blocking call to listdir, detected code that calls async_forward_entry_setup

### ISSUE THREAD

**Issue by garry0garry** [mrk-its/homeassistant-blitzortung#113]
FYI
After rebooting the Home Assistant.
```
homeassistant.util.loop
util/loop.py:136
Detected blocking call to listdir with args ('/config/deps/lib/python3.12/site-packages',) inside the event loop by custom integration 'blitzortung' at custom_components/blitzortung/mqtt.py, line 98: import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel (offender: /usr/local/lib/python3.12/importlib/metadata/__init__.py, line 680: return os.listdir(self.root or '.')), please create a bug report at https://github.com/mrk-its/homeassistant-blitzortung/issues For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#listdir Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 167, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 449, in _async_setup_component await asyncio.gather( File "/usr/src/homeassistant/homeassistant/setup.py", line 451, in <genexpr> create_eager_task( File "/usr/src/homeassistant/homeassistant/util/async_.py", line 37, in create_eager_task return Task(coro, loop=loop, name=name, eager_start=True) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 734, in async_setup_locked await self.async_setup(hass, integration=integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 586, in async_setup result = await component.async_setup_entry(hass, self) File "/config/custom_components/blitzortung/__init__.py", line 78, in async_setup_entry coordinator = BlitzortungCoordinator( File "/config/custom_components/blitzortung/__init__.py", line 201, in __init__ self.mqtt_client = MQTT( File "/config/custom_components/blitzortung/mqtt.py", line 98, in __init__ import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel
```

```
Detected code that calls async_forward_entry_setup for integration, blitzortung with title: Blitzortung and entry_id: daa33c5e05faa57db9815d06a68e68c3, which is deprecated and will stop working in Home Assistant 2025.6, await async_forward_entry_setups instead. Please report this issue.
Detected code that calls async_forward_entry_setup for integration blitzortung with title: Blitzortung and entry_id: daa33c5e05faa57db9815d06a68e68c3, during setup without awaiting async_forward_entry_setup, which can cause the setup lock to be released before the setup is done. This will stop working in Home Assistant 2025.1. Please report this issue.
```

**Comment by glcos** [comment]
Exact same issue here

**Comment by DeFlanko** [comment]
Same issue.. Logs told me to come here and report it... im doing my part:

```log
Please report this issue.
Stack (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once
    handle._run()
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2138, in async_forward_entry_setup
    report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 151, in report
    _LOGGER.warning(msg, stack_info=True)
```

**Comment by RedDaveNY** [comment]
same for me:
Detected blocking call to listdir with args ('/config/deps/lib/python3.12/site-packages',) inside the event loop by custom integration 'blitzortung' at custom_components/blitzortung/mqtt.py, line 98: import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel (offender: /usr/local/lib/python3.12/importlib/metadata/__init__.py, line 680: return os.listdir(self.root or '.')), please create a bug report at https://github.com/mrk-its/homeassistant-blitzortung/issues For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#listdir Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 167, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 449, in _async_setup_component await asyncio.gather( File "/usr/src/homeassistant/homeassistant/setup.py", line 451, in <genexpr> create_eager_task( File "/usr/src/homeassistant/homeassistant/util/async_.py", line 37, in create_eager_task return Task(coro, loop=loop, name=name, eager_start=True) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 734, in async_setup_locked await self.async_setup(hass, integration=integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 586, in async_setup result = await component.async_setup_entry(hass, self) File "/config/custom_components/blitzortung/__init__.py", line 78, in async_setup_entry coordinator = BlitzortungCoordinator( File "/config/custom_components/blitzortung/__init__.py", line 201, in __init__ self.mqtt_client = MQTT( File "/config/custom_components/blitzortung/mqtt.py", line 98, in __init__ import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel

**Comment by garry0garry** [comment]
Home Assistant 2024.8.2.
Blitzortung 1.0.1
```
Logger: homeassistant.util.loop
Source: util/loop.py:136
First occurred: August 22, 2024 at 9:45:40 PM (1 occurrences)
Last logged: August 22, 2024 at 9:45:40 PM

Detected blocking call to listdir with args ('/config/deps/lib/python3.12/site-packages',) inside the event loop by custom integration 'blitzortung' at custom_components/blitzortung/mqtt.py, line 98: import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel (offender: /usr/local/lib/python3.12/importlib/metadata/__init__.py, line 680: return os.listdir(self.root or '.')), please create a bug report at https://github.com/mrk-its/homeassistant-blitzortung/issues For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#listdir Traceback (most recent call last): File "<frozen runpy>", line 198, in _run_module_as_main File "<frozen runpy>", line 88, in _run_code File "/usr/src/homeassistant/homeassistant/__main__.py", line 223, in <module> sys.exit(main()) File "/usr/src/homeassistant/homeassistant/__main__.py", line 209, in main exit_code = runner.run(runtime_conf) File "/usr/src/homeassistant/homeassistant/runner.py", line 190, in run return loop.run_until_complete(setup_and_run_hass(runtime_config)) File "/usr/local/lib/python3.12/asyncio/base_events.py", line 674, in run_until_complete self.run_forever() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 641, in run_forever self._run_once() File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1990, in _run_once handle._run() File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run self._context.run(self._callback, *self._args) File "/usr/src/homeassistant/homeassistant/setup.py", line 165, in async_setup_component result = await _async_setup_component(hass, domain, config) File "/usr/src/homeassistant/homeassistant/setup.py", line 461, in _async_setup_component await asyncio.gather( File "/usr/src/homeassistant/homeassistant/setup.py", line 463, in <genexpr> create_eager_task( File "/usr/src/homeassistant/homeassistant/util/async_.py", line 37, in create_eager_task return Task(coro, loop=loop, name=name, eager_start=True) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 752, in async_setup_locked await self.async_setup(hass, integration=integration) File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup result = await component.async_setup_entry(hass, self) File "/config/custom_components/blitzortung/__init__.py", line 78, in async_setup_entry coordinator = BlitzortungCoordinator( File "/config/custom_components/blitzortung/__init__.py", line 201, in __init__ self.mqtt_client = MQTT( File "/config/custom_components/blitzortung/mqtt.py", line 98, in __init__ import paho.mqtt.client as mqtt # pylint: disable=import-outside-toplevel
```

**Comment by garry0garry** [comment]
Upd.
v1.0.1
![изображение](https://github.com/user-attachments/assets/1eec1717-3d1d-4ca0-996b-9b3e7df6513b)
v1.3.4
![изображение](https://github.com/user-attachments/assets/2978c824-5273-481e-b220-9fcd0994e6ac)

**Comment by mrk-its** [comment]
Fixed in 1.3.6
